ksp/src/main/kotlin/com/tschuchort/compiletesting/Ksp.kt

```diff
@@ -0,0 +1,68 @@
+/**
+ * Adds support for KSP (https://goo.gle/ksp).
+ */
+package com.tschuchort.compiletesting
+
+import org.jetbrains.kotlin.ksp.KotlinSymbolProcessingCommandLineProcessor
+import org.jetbrains.kotlin.ksp.KotlinSymbolProcessingComponentRegistrar
+import org.jetbrains.kotlin.ksp.processing.SymbolProcessor
+import java.io.File
+
+private const val KSP_PLUGIN_ID = "org.jetbrains.kotlin.ksp"
+
+private fun KotlinCompilation.initAndGetKspConfig(): KspConfiguration {
+    val config = KspConfiguration(workingDir.resolve("ksp"))
+    if (config.workingDir.exists()) {
+        // already configured, just return
+        return config
+    }
+    config.classesOutDir.mkdirs()
+    config.sourcesOurDir.mkdirs()
+    config.syntheticSources.mkdirs()
+    val kspOptions = listOf(
+        PluginOption(KSP_PLUGIN_ID, "apclasspath", config.syntheticSources.path),
+        PluginOption(KSP_PLUGIN_ID, "classes", config.classesOutDir.path),
+        PluginOption(KSP_PLUGIN_ID, "sources", config.sourcesOurDir.path)
+    )
+    compilerPlugins = compilerPlugins + KotlinSymbolProcessingComponentRegistrar()
+    commandLineProcessors = commandLineProcessors + listOf(KotlinSymbolProcessingCommandLineProcessor())
+    pluginOptions = pluginOptions + kspOptions
+    return config
+}
+
+fun KotlinCompilation.symbolProcessors(
+    vararg processors: Class<out SymbolProcessor>
+) {
+    check(processors.isNotEmpty()) {
+        "Must provide at least 1 symbol processor"
+    }
+    val config = initAndGetKspConfig()
+
+    // create a fake classpath that references our symbol processor
+    config.syntheticSources.apply {
+        resolve("META-INF/services/org.jetbrains.kotlin.ksp.processing.SymbolProcessor").apply {
+            parentFile.mkdirs()
+            // keep existing ones in case this function is called multiple times
+            val existing = if (exists()) {
+                readLines(Charsets.UTF_8)
+            } else {
+                emptyList()
+            }
+            val processorNames = existing + processors.map {
+                it.typeName
+            }
+            writeText(
+                processorNames.joinToString(System.lineSeparator())
+            )
+        }
+    }
```

Comment:
So you are writing all the class names in a file so they can be discovered by a service locator later on? Is there no better way? Why not make your own `SymbolProcessor` that is declared as a service and has a list of the user's processors, similar to `MainComponentRegistrar`?

Reply:
I didn't want to do that because it won't necessarily match the way KSP ends up running them. 
e.g. for instance, KSP might start doing compilations in between processors, or giving them different instances of resolvers, loggers etc.
Idk if that'll ever happen but i thought keeping it as close to real as possible is a better approach.
wdyt?

Reply:
Hmm, I hadn't thought about that. I need to take a closer look tomorrow. Maybe @neetopia can comment if it would make a difference?

Reply:
Currently we don't create multiple instances of resolvers, but it is possible if we support multiple round processing.